pkg/services/user/userimpl/store.go

```diff
@@ -490,8 +494,8 @@ func (ss *sqlStore) Search(ctx context.Context, query *user.SearchUsersQuery) (*
 		whereParams = append(whereParams, acFilter.Args...)
 
 		if query.Query != "" {
-			whereConditions = append(whereConditions, "(email "+ss.dialect.LikeStr()+" ? OR name "+ss.dialect.LikeStr()+" ? OR login "+ss.dialect.LikeStr()+" ?)")
-			whereParams = append(whereParams, queryWithWildcards, queryWithWildcards, queryWithWildcards)
+			whereConditions = append(whereConditions, "(u.email "+ss.dialect.LikeStr()+" ? OR u.name "+ss.dialect.LikeStr()+" ? OR u.login "+ss.dialect.LikeStr()+" ? OR org_user.role "+ss.dialect.LikeStr()+" ?)")
+			whereParams = append(whereParams, queryWithWildcards, queryWithWildcards, queryWithWildcards, queryWithWildcards)
```

Comment:
Thank you for your comments! I agree that it can be quite difficult for an administrator to work with the case when one user belongs to several organizations. This requires a reworking of the user list displaying itself, not only a search and filtering.

In this case, I can remove the current changes from `All Users`, they will remain as is, and for searching by organization I could add a filter for the role, but leave the search as it is now (for login, name and email). In this way, we will, firstly, be able to localize the search for the user by the combination of login details and role, and secondly, we will provide an uncluttered search for all possible user columns in the UI.